File: api/check_db.php
<?php
header('Content-Type: application/json');
header('Access-Control-Allow-Origin: *');

try {
    $dbFile = __DIR__ . '/nexus_gpt.db';
    $db = new PDO("sqlite:$dbFile");
    $db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
    
    // Check storage table for history entries
    $stmt = $db->query("SELECT username, key_name, LENGTH(value_data) as data_length FROM storage WHERE key_name LIKE 'nexus_history_%' ORDER BY username, key_name");
    $historyRows = $stmt->fetchAll(PDO::FETCH_ASSOC);
    
    // Check all users
    $userStmt = $db->query("SELECT username, role, email, phone FROM users ORDER BY username");
    $users = $userStmt->fetchAll(PDO::FETCH_ASSOC);
    
    // Sample history data (first user)
    $sampleHistory = null;
    if (!empty($historyRows)) {
        $firstUser = $historyRows[0]['username'];
        $sampleStmt = $db->prepare("SELECT value_data FROM storage WHERE username = ? AND key_name = ?");
        $sampleStmt->execute([$firstUser, 'nexus_history_' . $firstUser]);
        $sampleData = $sampleStmt->fetchColumn();
        $sampleHistory = json_decode($sampleData, true);
    }
    
    echo json_encode([
        'success' => true,
        'db_exists' => file_exists($dbFile),
        'db_size' => file_exists($dbFile) ? filesize($dbFile) : 0,
        'history_entries' => count($historyRows),
        'history_rows' => $historyRows,
        'users' => $users,
        'sample_history_sessions' => $sampleHistory ? array_keys($sampleHistory) : [],
        'sample_session_count' => $sampleHistory ? count($sampleHistory) : 0
    ], JSON_PRETTY_PRINT);
    
} catch (Exception $e) {
    echo json_encode(['success' => false, 'error' => $e->getMessage()]);
}
?>

File: check_db.py
import sqlite3
import json
import os

os.chdir(r'C:\Users\SERVER PC\Downloads\NEXUS GPT')

db_path = r'server/nexus_gpt.db'
print('Database path:', os.path.abspath(db_path))
print('DB exists:', os.path.exists(db_path))

if os.path.exists(db_path):
    conn = sqlite3.connect(db_path)
    conn.row_factory = sqlite3.Row
    cursor = conn.cursor()
    
    # Check tables
    cursor.execute("SELECT name FROM sqlite_master WHERE type='table'")
    tables = [row['name'] for row in cursor.fetchall()]
    print('\nTables:', tables)
    
    # Check users
    if 'users' in tables:
        cursor.execute('SELECT username, role, email FROM users')
        users = cursor.fetchall()
        print('\nUsers:')
        for u in users:
            print(f'  - {u["username"]} ({u["role"]}) - {u["email"]}')
    
    # Check history entries in storage
    if 'storage' in tables:
        cursor.execute("SELECT username, key_name, LENGTH(value_data) as size FROM storage WHERE key_name LIKE 'nexus_history_%'")
        history = cursor.fetchall()
        print(f'\nHistory entries found: {len(history)}')
        for h in history:
            print(f'  - User: {h["username"]}, Key: {h["key_name"]}, Size: {h["size"]} bytes')
        
        # Sample history content
        if history:
            first_user = history[0]['username']
            cursor.execute('SELECT value_data FROM storage WHERE username=? AND key_name=?', 
                         [first_user, f'nexus_history_{first_user}'])
            data = cursor.fetchone()
            if data:
                try:
                    parsed = json.loads(data['value_data'])
                    print(f'\nSample history for {first_user}: {len(parsed)} sessions')
                    for sid in list(parsed.keys())[:3]:
                        print(f'  - Session: {sid[:30]}...')
                except Exception as e:
                    print(f'Could not parse history JSON: {e}')
    
    conn.close()
else:
    print('Database file not found!')
